apollo-federation/tests/query_plan/build_query_plan_tests/defer.rs

```diff
@@ -6,71 +6,6 @@ fn config_with_defer() -> QueryPlannerConfig {
     config
 }
 
-#[test]
-fn defer_test_handles_simple_defer_without_defer_enabled() {
```

Comment:
Is there a reason this was removed? It was ported directly from JS, and, IMO, is testing behavior that we care about.

Reply:
The changed behavior does not allow `@defer` directive if defer is enabled, while previously `@defer` was removed silently. So, the tested scenario can no longer happen.

However, I just noticed the test would still be able to run. I think this function need to be updated:
https://github.com/apollographql/router/blob/6342406c70edcefbc62bbb6c37bd397a4dc080b9/apollo-federation/tests/query_plan/build_query_plan_support.rs#L98
```rust
pub(crate) fn api_schema_and_planner(
    function_path: &'static str,
    config: QueryPlannerConfig,
    subgraph_names_and_schemas: &[(&str, &str)],
) -> (ValidFederationSchema, QueryPlanner) {
    let supergraph = compose(function_path, subgraph_names_and_schemas);
    let supergraph = apollo_federation::Supergraph::new(&supergraph).unwrap();
    let planner = QueryPlanner::new(&supergraph, config).unwrap();
    let api_schema_config = apollo_federation::ApiSchemaOptions {
        include_defer: true,
        include_stream: false,
    };
    let api_schema = supergraph.to_api_schema(api_schema_config).unwrap();
    (api_schema, planner)
}
```
The config should control `ApiSchemaOptions`'s `include_defer`.

Reply:
thats the thing -> outside of tests it is not possible for this scenario to ever reach QP. Old test was verifying the behavior that we would strip out `@defer` from incoming requests if it was disabled.... but if user explicitly opts out of `@defer` but still send a query with `@defer` it will be rejected during GraphQL validation by `apollo-rs` (as that directive won't exist in the API schema).

So real use cases are
- defer enabled
  - query with defer (need to normalize it)
  - query without defer 
- defer disabled
  - query without defer

Reply:
Fixed the test api schema config in https://github.com/apollographql/router/pull/6470/commits/6d7a1ca018e9ba35ab8b6937bb4fb9112269352d